sdk/documenttranslation/azure-ai-documenttranslation/tests/testcase.py

```diff
@@ -113,6 +120,148 @@ def generate_sas_url(self, container_name, permission):
+        self.assertIsNotNone(doc_details.id)
+        self.assertIsNotNone(doc_details.translated_document_url)
+        self.assertIsNotNone(doc_details.translation_progress)
+        self.assertIsNotNone(doc_details.characters_charged)
+        self.assertIsNotNone(doc_details.created_on)
+        self.assertIsNotNone(doc_details.last_updated_on)
+
+    def _validate_translation_job(self, job_details, **kwargs):
+        status = kwargs.pop("status", None)
+        total = kwargs.pop('total', None)
+        failed = kwargs.pop('failed', None)
+        succeeded = kwargs.pop('succeeded', None)
+        inprogress = kwargs.pop('inprogress', None)
+        notstarted = kwargs.pop('notstarted', None)
+        cancelled = kwargs.pop('cancelled', None)
+        # status
+        self.assertEqual(job_details.status, status) if status else self.assertIsNotNone(job_details.status)
+        # docs count
+        self.assertEqual(job_details.documents_total_count, total) if total else self.assertIsNotNone(job_details.documents_total_count)
+        self.assertEqual(job_details.documents_failed_count, failed) if failed else self.assertIsNotNone(job_details.documents_failed_count)
+        self.assertEqual(job_details.documents_succeeded_count, succeeded) if succeeded else self.assertIsNotNone(job_details.documents_succeeded_count)
+        self.assertEqual(job_details.documents_in_progress_count, inprogress) if inprogress else self.assertIsNotNone(job_details.documents_in_progress_count)
+        self.assertEqual(job_details.documents_not_yet_started_count, notstarted) if notstarted else self.assertIsNotNone(job_details.documents_in_progress_count)
+        self.assertEqual(job_details.documents_cancelled_count, cancelled) if cancelled else self.assertIsNotNone(job_details.documents_cancelled_count)
+        # generic assertions
+        self.assertIsNotNone(job_details.id)
+        self.assertIsNotNone(job_details.created_on)
+        self.assertIsNotNone(job_details.last_updated_on)
+        self.assertIsNotNone(job_details.total_characters_charged)
+
+    def _validate_format(self, format):
+        #self.assertIsNotNone(format.format)
+        self.assertIsNotNone(format.file_extensions)
+        self.assertIsNotNone(format.content_types)
+        #self.assertIsNotNone(format.format_versions)
+
+
+    # client helpers
+    def _submit_and_validate_translation_job(self, client, translation_inputs, total_docs_count):
+        # submit job
+        job_details = client.create_translation_job(translation_inputs)
+        self.assertIsNotNone(job_details.id)
+        # wait for result
+        job_details = client.wait_until_done(job_details.id)
+        # validate
+        self._validate_translation_job(job_details=job_details, status='Succeeded', total=total_docs_count, succeeded=total_docs_count)
+
+        return job_details.id
+
+    async def _submit_and_validate_translation_job_async(self, async_client, translation_inputs, total_docs_count):
```

Comment:
one last thing - create an asynctestcase.py for these helper validation methods that use async. Python 2.7 won't be able to read this file and sees any async code as a SyntaxError.